soroban-env-host/src/host.rs

```diff
@@ -890,6 +867,38 @@ impl Host {
 
         Ok(())
     }
+
+    // If autobump enabled, autobumps all the entries in the footprint.
+    fn maybe_autobump_expiration(&self) -> Result<(), HostError> {
+        let Some(autobump_ledgers) = self.with_ledger_info(|li| {
+            if li.autobump_ledgers > 0 {
+                Ok(Some(li.autobump_ledgers))
+            } else {
+                Ok(None)
+            }
+        })? else {
+            return Ok(());
+        };
+        // Need to copy the footprint out of the storage to allow mut borrow of
+        // storage.
+        let footprint = self.try_borrow_storage()?.footprint.clone();
```

Comment:
Switched to metered clone. IIUC extracting the keys to bump would require allocating a new container, which seems trickier from the metering standpoint, so not sure if it's worth the marginal savings for certain footprints.

> you can skip doing the has lookup for any part of the footprint that's AccessType::ReadOnly since those can never be written to storage.

I don't think that would be correct - RO entries don't actually have to be present in storage (one more benefit of performing the bump logic in storage is to correctly handle bumping the missing entries).